### ISSUE TITLE
Kaleido 0.2.1 can  not start in Debian GNU/Linux 12 (bookworm) with Ploty 5.20.0

### ISSUE THREAD

**Issue by BenLo00** [plotly/Kaleido#175]
I am using a plotly plot in rasperian (Debian GNU/Linux 12 (bookworm)) and I want to convert it with Kaleido to a PNG,
wich worked on older Hardware/Software fine but dos not in this environment. if I invoke Kaleido it drops the error

File "/home/pi/tempueberwachung.py", line 234, in <module>
    raise error
  File "/home/pi/tempueberwachung.py", line 182, in <module>
    fig_bytes = fig.to_image(format='png', width=1920, height=1080)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/plotly/basedatatypes.py", line 3778, in to_image
    return pio.to_image(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/plotly/io/_kaleido.py", line 143, in to_image
    img_bytes = scope.transform(
                ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/kaleido/scopes/plotly.py", line 153, in transform
    response = self._perform_transform(
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/kaleido/scopes/base.py", line 293, in _perform_transform
    self._ensure_kaleido()
  File "/usr/local/lib/python3.11/dist-packages/kaleido/scopes/base.py", line 198, in _ensure_kaleido
    raise ValueError(message)
ValueError: Failed to start Kaleido subprocess. Error stream:

/usr/local/lib/python3.11/dist-packages/kaleido/executable/kaleido: line 10: 10785 Segmentation fault      ./bin/kaleido $@

Does someone have a clue? do you need more infos about my environment? any advise welcome!
Greetings, Ben.

**Comment by bedeberger** [comment]
Same error here: https://github.com/plotly/Kaleido/issues/59

**Comment by gvwilson** [comment]
Thanks for your interest in Kaleido. We are currently working on an overhaul that might address your issue - we hope to have news in a few weeks and will post an update then. Thanks - @gvwilson

**Comment by gwild** [comment]
Any update?  Kaleido still causes a segmentation fault as of today.

**Comment by ayjayt** [comment]
@gwild are you also working on rasperian?

**Comment by gwild** [comment]
Yes, Raspian Bookworm on Pi5

**Comment by gwild** [comment]
Bookworm forces use of virtual environment.  Path to Kaleido directory is broken for virtual environments.

**Comment by ayjayt** [comment]
@gwild, there a pre-lease of `kaleido2`  https://pypi.org/project/kaleido/1.0.0rc0/ out which can be installed with `pip install --pre kaleido`.

It is the precursor for what will be `pip install kaleido2`.

It allows to use an external chromium interface instead of the `kaleido` executable. I am not super familiar with bookworm, but you might try using that version. It requires you to a) have `chrome` or `chromium` in path or you can set the environmental variable `BROWSER_PATH`.

That may be the best direction for you- if not, I will have to learn a little bit more about whats going on with raspian and bookworm.

Out of curiosity, is this for your spectrum analyzer that I see on your profile?

**Comment by gwild** [comment]
Yes, for spectrum analyzer, though not the rust version in my profile.  I’m using  plotly instead of matplotlib because this is a 24/7 script and matplotlib leaks memory.

**Comment by gwild** [comment]
I was able to get it to run by editing Dir variable in Kaleido to the actual path.
